### ISSUE TITLE
Ability to reset Payment Request button on product page

### ISSUE THREAD

**Issue by asumaran** [woocommerce/woocommerce-gateway-stripe#1624]
### The Problem

While working #1563 we found issues (e.g. https://github.com/woocommerce/woocommerce-gateway-stripe/issues/1599, p1624330024334600-slack-C07414X4N) when working on adding support for more complex product types. Currently, the Payment Request button is initialized when the product page loads. So, if the loaded product required shipping the PRB was initialized with `requestShipping: true` ([Stripe Docs](https://stripe.com/docs/js/payment_request/create#stripe_payment_request-options-requestShipping)).

This works well for simple product types since the "Shipping" status don't change later on but then when working with Bundles/Composite product types the case seems more diverse. The product(s) the user ends up purchasing may end up wether requiring Shipping or not.

For these product types the status of Shipping depends on user interaction. We can't know before hand if the PRB has to be initialized with `requestShipping` set to `true` or `false`.

### How to fix

The most reliable way to know if a product will require Shipping is to add the product to the Cart and return the result of `WC()->cart->needs_shipping()` to the frontend. Then in the frontend we can destroy the current PR button and initialize a new one with an updated value for `requestShipping`.

<img src="https://user-images.githubusercontent.com/1025173/124517274-30086480-dda9-11eb-8048-902c50932c25.png" width="497" />

This way we can show/hide the "Shipping Address" section depending on the status of `WC()->cart->needs_shipping()` accordingly.

The flow would be something like this.

1. User visits product page
2. Requests to get PRB config object
3. PRB is initialized
4. User clicks PRB
5. Request to backend adding product to cart
6. Get response from adding product to cart
7. Destroy current PRB then initialize new one using config from previous response
8. Show browser's payment modal.

Note: Not sure if we will be able to destroy, then create a PRB **and** open the browser's payment modal programmatically. If this is not possible then we need to find another approach in order to support complex support types.

### Downsides

- PRB button will be removed and then inserted to the page so it might cause some flicks on the page
- Possible delay since Stripe could make some requests in order to create a PRB.

### Benefits

- Enable support for Bundles and Composite product types.
- Basically we will be able to add support for any product type compatible with WooCommerce's Cart.
